src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/EventConsumerIntegrationTest.java

```diff
@@ -0,0 +1,272 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Schedulers;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.ServiceBusConfiguration;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.executors.CcdCaseEventsDeadLetterQueueExecutor;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.executors.CcdCaseEventsExecutor;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.executors.CcdEventExecutor;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.executors.CcdMessageProcessorExecutor;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.executors.MessageReadinessExecutor;
+import uk.gov.hmcts.reform.wacaseeventhandler.entity.CaseEventMessageEntity;
+import uk.gov.hmcts.reform.wacaseeventhandler.repository.CaseEventMessageRepository;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.annotation.DirtiesContext.ClassMode.AFTER_CLASS;
+import static org.springframework.test.annotation.DirtiesContext.HierarchyMode.EXHAUSTIVE;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+@SpringBootTest
+@ActiveProfiles(profiles = {"db", "integration"})
+@ContextConfiguration(classes = EventConsumerIntegrationTest.TestConfig.class)
+@TestPropertySource(properties = {"azure.servicebus.enableASB=true",
+    "azure.servicebus.connection-string="
+        + "Endpoint=sb:test;SharedAccessKeyName=test;SharedAccessKey=test;EntityPath=test",
+    "azure.servicebus.topic-name=test",
+    "azure.servicebus.subscription-name=test",
+    "azure.servicebus.ccd-case-events-subscription-name=test",
+    "azure.servicebus.retry-duration=2"})
+@DirtiesContext(classMode = AFTER_CLASS, hierarchyMode = EXHAUSTIVE)
```

Comment:
Still need this?

Reply:
Yeah, Still needed. Without this spring carry on to the other test with the same context and with consumers keep running.